dart/dynamics/InverseKinematics.cpp

```diff
@@ -1078,7 +1090,8 @@ void InverseKinematics::Analytical::computeGradient(
       mIK->getErrorMethod().computeDesiredTransform(
         mIK->getNode()->getWorldTransform(), _error);
 
-  if(PRE_ANALYTICAL == mAnalyticalP.mExtraDofUtilization
+  if((PRE_ANALYTICAL == mAnalyticalP.mExtraDofUtilization
+      || PRE_AND_POST_ANALYTICAL == mAnalyticalP.mExtraDofUtilization)
```

Comment:
The Analytical base class has an option for how to make use of any DOFs which aren't directly used by the Analytical solver. They'll be used in a gradient descent based on their Jacobian transpose, and this can either be done before the Analytical solution is computed (``PRE_ANALYTICAL``), after the Analytical solution is computed (``POST_ANALYTICAL``) or both (``PRE_AND_POST_ANALYTICAL``).

However, as I mention in my top-level comment, I would very much like to redesign this approach to utilizing the extra DOFs. I don't have any particularly compelling ideas right now (nor do I expect to have any in the immediate future, since I'm tied up with thesis work), but I'd be happy to discuss any ideas that anyone has.